md-dao/src/main/java/com/wipro/ats/bdre/md/dao/LineageQueryDAO.java

```diff
@@ -49,6 +52,45 @@
         return lineageQuerys;
     }
 
+    //get Instance exec ids for the process id from LQ table
+    private Long getInstanceExecIds(Integer processId) {
+        Long instanceExecId = null;
+        Session session = sessionFactory.openSession();
+        session.beginTransaction();
+        Criteria getLastElementCriteria = session.createCriteria(LineageQuery.class).add(Restrictions.eq("processId", processId)).addOrder(Order.desc("instanceExecId"));
+
+        if(getLastElementCriteria.list().size() != 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1155)